From 5f29e42b563b7b5975948720135ec624fdcc7763 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <lass@mail.uni-paderborn.de>
Date: Thu, 19 Feb 2015 19:34:00 +0100
Subject: [PATCH 1/4] Adjust RedHat config and service files

Basically we reuse upstream's config for RedHat but we change a few things:

 * adjust paths to match ArchLinux packaging
 * remove the dynamic creation of CellServDB
 * enable network traffic encryption by default
 * wait for dkms if dkms is used

Change-Id: Iddf101a047458dda34a000dec62feefd81c13421
---
 src/packaging/RedHat/openafs-cacheinfo      |  2 +-
 src/packaging/RedHat/openafs-client.service | 17 ++++++++---------
 src/packaging/RedHat/openafs-server.service |  4 ++--
 3 files changed, 11 insertions(+), 12 deletions(-)

diff --git a/src/packaging/RedHat/openafs-cacheinfo b/src/packaging/RedHat/openafs-cacheinfo
index dd0fd05cb..1663c79b5 100755
--- a/src/packaging/RedHat/openafs-cacheinfo
+++ b/src/packaging/RedHat/openafs-cacheinfo
@@ -1 +1 @@
-/afs:/usr/vice/cache:100000
+/afs:/var/cache/openafs:100000
diff --git a/src/packaging/RedHat/openafs-client.service b/src/packaging/RedHat/openafs-client.service
index 09379bea3..a63fce486 100644
--- a/src/packaging/RedHat/openafs-client.service
+++ b/src/packaging/RedHat/openafs-client.service
@@ -1,21 +1,20 @@
 [Unit]
 Description=OpenAFS Client Service
 Wants=network-online.target
-After=syslog.target network-online.target
+After=syslog.target network-online.target dkms.service
 Before=remote-fs.target
 
 [Service]
 Type=forking
 RemainAfterExit=true
-EnvironmentFile=/etc/sysconfig/openafs
+EnvironmentFile=/etc/conf.d/openafs
 ExecStartPre=/bin/bash -c "fs sysname > /dev/null 2>/dev/null; test $? -ne 0 || (echo AFS client appears to be running -- not starting && exit 1)"
-ExecStartPre=/bin/sed -n 'w/usr/vice/etc/CellServDB' /usr/vice/etc/CellServDB.local /usr/vice/etc/CellServDB.dist
-ExecStartPre=/bin/chmod 0644 /usr/vice/etc/CellServDB
-ExecStartPre=/sbin/modprobe openafs
-ExecStart=/usr/vice/etc/afsd $AFSD_ARGS
-ExecStop=/bin/umount /afs
-ExecStop=/usr/vice/etc/afsd -shutdown
-ExecStop=/sbin/rmmod openafs
+ExecStartPre=/usr/bin/modprobe openafs
+ExecStart=/usr/bin/afsd $AFSD_ARGS
+ExecStartPost=/usr/bin/fs setcrypt on
+ExecStop=/usr/bin/umount /afs
+ExecStop=/usr/bin/afsd -shutdown
+ExecStop=/usr/bin/rmmod openafs
 KillMode=none
 
 [Install]
diff --git a/src/packaging/RedHat/openafs-server.service b/src/packaging/RedHat/openafs-server.service
index ac9d3c411..0a4b3bfea 100644
--- a/src/packaging/RedHat/openafs-server.service
+++ b/src/packaging/RedHat/openafs-server.service
@@ -3,8 +3,8 @@ Description=OpenAFS Server Service
 After=syslog.target network.target
 
 [Service]
-EnvironmentFile=-/etc/sysconfig/openafs
-ExecStart=/usr/afs/bin/bosserver -nofork $BOSSERVER_ARGS
+EnvironmentFile=-/etc/conf.d/openafs
+ExecStart=/usr/bin/bosserver -nofork $BOSSERVER_ARGS
 ExecStop=/usr/bin/bos shutdown localhost -wait -localauth
 
 [Install]
-- 
2.15.1

